### PR TITLE
update EntityFactory types + compiling errors

### PR DESCRIPTION
**Pull request by nelsonsbrian**
Did some EntityFactory types on my repo, so transferred the work I did to here.

There is some wonky assumptions made between the property of id/entityReference/area between the Area + Item/Npc/Room that some of them use some of those properties, and some others.

I think this gets us 1 step closer to true types and less type bypassing.

### CODE REVIEW

**Comment by nelsonsbrian** [comment]
Also updated some types that were failing on build

File: src/Npc.ts [hunk 1]
```diff
@@ -6,25 +6,26 @@ import { IGameState } from './GameState';
...
 	description: string;
 	entityReference: EntityReference;
-	id: string | number;
+	id: string;
 	keywords: string[];
 	quests?: EntityReference[];
 	uuid?: string;
 }
 
-export class Npc extends Scriptable(Character) {
```
Comment (nelsonsbrian) [inline src/Npc.ts:27]: Scriptable is already in Character and this was causing some circular dependency issues I believe

File: package.json [hunk 2]
```diff
@@ -1,51 +1,51 @@
...
+		"./dist/esm"
+	],
+	"dependencies": {
+		"@types/bcryptjs": "^2.4.2",
+		"@types/js-yaml": "^3.12.5",
+		"@types/uuid": "^8.3.0",
+		"@types/winston": "^2.4.4",
+		"@types/ws": "^7.4.5",
+		"bcryptjs": "^2.4.0",
+		"commander": "^4.1.0",
```
Comment (nelsonsbrian) [inline package.json:35]: had a compile error from commander in the node_modules. We only use the 'types' in core-ts, so just upgraded to the real package (as the types are just a stub) until the error went away.

File: src/Npc.ts (hunk 1 above)
Comment (seanohue) [reply src/Npc.ts:27]: I think you're right about this -- I forget if I fixed this the same way locally but I definitely ran into it. Good call

File: src/BundleManager.ts [hunk 3]
```diff
@@ -376,11 +378,14 @@ export class BundleManager {
 	 * @param {EntityFactory} factory
 	 * @return {Array<string>}
 	 */
-	async loadEntities(
+	async loadEntities<
```
Comment (seanohue) [inline src/BundleManager.ts:381]: good use of generics here

**Review by seanohue — APPROVED** [review]

**Pull request merged**